src/lib.rs

```diff
@@ -53,101 +53,153 @@ use indexmap::IndexMap;
+/// Take a box as a string and return the encoded bytes representing the box type-format.
 // Note: box refers to an SSB `private box` and not a Rust `Box` (pointer type)
-pub fn get_box_type(box_str: &str) -> Result<Vec<u8>> {
+pub fn get_box_type(boxed_str: &str) -> Result<Vec<u8>> {
     let box_type;
-    if box_str.ends_with(".box") {
-        box_type = BOX_1_TYPE
-    } else if box_str.ends_with(".box2") {
-        box_type = BOX_2_TYPE
+    if boxed_str.ends_with(".box") {
+        box_type = BOX1_TF
+    } else if boxed_str.ends_with(".box2") {
+        box_type = BOX2_TF
     } else {
-        return Err(anyhow!("The box type is unknown"));
+        return Err(anyhow!("Unknown boxed string: {}", boxed_str));
     };
 
     Ok(box_type.to_vec())
 }
 
-/// Take a feed identity (key) as a string and return the encoded bytes representing the feed type.
-pub fn get_feed_type(feed: &str) -> Result<Vec<u8>> {
+/// Take a feed ID (key) as a string and return the encoded bytes representing the feed type-format.
+pub fn get_feed_type(feed_id: &str) -> Result<Vec<u8>> {
     let feed_type;
-    if feed.ends_with(".ed25519") {
-        feed_type = CLASSIC_FEED_TYPE
-    } else if feed.ends_with(".bbfeed-v1") {
-        feed_type = BB_FEED_TYPE
-    } else if feed.ends_with(".ggfeed-v1") {
-        feed_type = GG_FEED_TYPE
+    if feed_id.ends_with(".ed25519") {
+        feed_type = CLASSIC_FEED_TF
+    } else if feed_id.ends_with(".bbfeed-v1") {
+        feed_type = BENDYBT_FEED_TF
+    } else if feed_id.ends_with(".ggfeed-v1") {
+        feed_type = GABBYGR_FEED_TF
     } else {
-        return Err(anyhow!("The feed type is of an unknown format"));
+        return Err(anyhow!("Unknown feed format: {}", feed_id));
     };
 
     Ok(feed_type.to_vec())
 }
 
-/// Take a message key as a string and return the encoded bytes representing the message type.
-pub fn get_msg_type(msg: &str) -> Result<Vec<u8>> {
+/// Take a message ID as a string and return the encoded bytes representing the message type-format.
+pub fn get_msg_type(msg_id: &str) -> Result<Vec<u8>> {
     let msg_type;
-    if msg.ends_with(".sha256") {
-        msg_type = CLASSIC_MSG_TYPE
-    } else if msg.ends_with(".bbmsg-v1") {
-        msg_type = BB_MSG_TYPE
-    } else if msg.ends_with(".ggmsg-v1") {
-        msg_type = GG_MSG_TYPE
+    if msg_id.ends_with(".sha256") {
+        msg_type = CLASSIC_MSG_TF
+    } else if msg_id.ends_with(".bbmsg-v1") {
+        msg_type = BENDYBT_MSG_TF
+    } else if msg_id.ends_with(".ggmsg-v1") {
+        msg_type = GABBYGR_MSG_TF
     } else {
-        return Err(anyhow!("The message type is unknown"));
+        return Err(anyhow!("Unknown message ID: {}", msg_id));
     };
 
     Ok(msg_type.to_vec())
 }
 
+/// Take a blob ID as a string and return the encoded bytes as a vector.
+pub fn encode_blob(blob_id: &str) -> Result<Vec<u8>> {
+    let mut encoded_blob = get_blob_type(blob_id)?;
```

Comment:
(Help me remember, I'm not great at predicting how Rust will behave:) will this `mut` mean that `CLASSIC_BLOB_TF` will be mutated (that would be a problem), or does `get_blob_type` make a copy of `CLASSIC_BLOB_TF` before returning?

Reply:
The `mut` means that the value of `encoded_blob` can be mutated after the initial assignment. Mutability is required in this case because we're appending bytes to the `encoded_blob` `Vec` using `base64::decode_config_buf` (L193).

An alternative approach would be to assign `encoded_blob` without mutability and then use `base64::decode_config` to return a new `Vec<u8>` with the decoded substring bytes. We would then have to concat the `encoded_blob` `Vec` with the newly-created substring bytes `Vec`.

[Base64 decode docs here](https://docs.rs/base64/0.13.0/base64/#decoding).

Reply:
Okay, but that doesn't mutate `CLASSIC_BLOB_TF`, right? I'm only concerned that mutation in this function would persist to other cases in the future

Reply:
Nope, `CLASSIC_BLOB_TF` is a `const` and cannot be mutated. The only value that can possibly be mutated in this scope (`fn encode_blob()`) is `encoded_blob` (which is an owned value, ie. not a reference, returned by `fn get_blob_type()`). It is mutated once through the appending of the `base64_str` bytes.